lib/daru/io/base.rb

```diff
@@ -0,0 +1,16 @@
+module Daru
+  module IO
+    class Base
+      def initialize(binding)
+        args = method(__method__).parameters.map do |_, name|
+          [name, binding.local_variable_get(name.to_s)]
+        end.to_h
+
+        args.each do |k, v|
+          instance_variable_set("@#{k}", v)
+          define_singleton_method(k) { instance_variable_get("@#{k}") }
+        end
+      end
```

Comment:
OK, I understand I have not been here at the time of merging, but I should say I _highly_ disapprove on this "black magic" with binding. Our tasks definitely are NOT that complicated to justify this amount of metaprogramming.